source/libraries/ContractInterfaces.ts

```diff
@@ -34,8 +34,9 @@
         protected async remoteCall(abi: AbiFunction, parameters: Array<any>, txName: String, sender?: string, gasPrice?: BN, attachedEth?: BN): Promise<void> {
             const from = sender || this.accountManager.defaultAddress;
             const data = encodeMethod(abi, parameters);
-            // TODO: remove `gas` property once https://github.com/ethereumjs/testrpc/issues/411 is fixed
-            const gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data }, attachedEth ? { value: attachedEth } : {} ));
+            let gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data }, attachedEth ? { value: attachedEth } : {} ));
+            // This is to address an observed bug while running against geth where occasionally gas estimates are lower than required
```

Comment:
Once we have a repro case in a Geth GitHub issue, we should link to it in this comment so a future reader can know whether it is safe to delete this terrible code.